### PR TITLE
Be compatible with mimic-happy-eyeballs.0.0.9

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

File: test/test.ml [hunk 1]
```diff
@@ -149,7 +166,7 @@ let test01 =
     server ~stop (Tcpip_stack_socket.V4V6.tcp stack) (`HTTP_1_1 (8080, handler))
   in
   let* result =
-    Http_mirage_client.request t "http://localhost:8080/"
+    Http_mirage_client.request t "http://127.0.0.1:8080/"
```
Comment (hannesm) [inline test/test.ml:169]: why this change? does name resolution no longer work?

File: test/test.ml [hunk 2]
```diff
@@ -197,7 +214,7 @@ let test02 =
   in
   let str = random_string ~len:0x1000 in
   let* result =
-    Http_mirage_client.request ~meth:`POST ~body:str t "http://localhost:8080/"
+    Http_mirage_client.request ~meth:`POST ~body:str t "http://127.0.0.1:8080/"
```
Comment (hannesm) [inline test/test.ml:217]: same as above, can we get dns resolution back in the test? :)

Comment (dinosaure) [reply test/test.ml:217]: We actually don't inject the `getaddrinfo` (and a DNS instance) into our happy-eyeballs instance. Note that for the mirage tool, we do that via `functoria` (and the `connect` function). For the testing purpose, we just would like to see if we are able to send an HTTP request to a server (with or without DNS). To avoid the DNS request (and the injection of a DNS instance), we just need to use 127.0.0.1 instead of `localhost`.
